nixos/doc/manual/release-notes/rl-2305.section.md

```diff
@@ -67,6 +67,8 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - In `mastodon` it is now necessary to specify location of file with `PostgreSQL` database password. In `services.mastodon.database.passwordFile` parameter default value `/var/lib/mastodon/secrets/db-password` has been changed to `null`.
 
+- The `--target-host` and `--build-host` options of `nixos-rebuild` don't treat the `localhost` value special anymore. The empty string must now be used to force a local deployment/build.
```

Comment:
The empty string is default so there is no need to force anything with this change.

Reply:
The empty string is now the only way to reset to local behavior.

I was thinking about cases where `nixos-rebuild` is wrapped by another script (which is what I do), which pre-fills those options and the user might want to restore the default.

The script might contain something like this:

```shell
exec nixos-rebuild --build-host builder --flake myconfigs#host "$@"
```

Reply:
How about this text:

> The `--target-host` and `--build-host` options of `nixos-rebuild` don't treat the `localhost` value special anymore. Local operation is now represented by the empty string.

Or should the "empty string" part simply be removed?

Reply:
Maybe something like:

> Specifying a target host without a build host will not default the latter to the former. Additionally, the `--target-host` and `--build-host` options of `nixos-rebuild` no longer treat the `localhost` value specially – to build on/deploy to local machine, omit the relevant flag.

Reply:
> Specifying a target host without a build host will not default the latter to the former

This wasn't the case before #148921, both values defaulted to local, even if only one was specified.
So this was also true before:

> to build on/deploy to local machine, omit the relevant flag.

Reply:
> > Specifying a target host without a build host will not default the latter to the former
> 
> This wasn't the case before #148921, both values defaulted to local, even if only one was specified.

Oh, you are right. It was true when I created my deployment script but since then https://github.com/NixOS/nixpkgs/commit/3d34fe9f2b55e29721fa0c0c8ad677c0bf064e3e changed the `buildHost` default.

The second sentence should be sufficient then.

Reply:
> It was true when I created my deployment script but since then [3d34fe9](https://github.com/NixOS/nixpkgs/commit/3d34fe9f2b55e29721fa0c0c8ad677c0bf064e3e) changed the `buildHost` default.

I see, this was changed before. I must have build my script afterwards, because I can't remember having this problem.

I updated the changelog with your suggested text, thanks.